Source/Relativity.Test.Helpers/Relativity.Test.Helpers/Kepler/OAuth2Helper.cs

```diff
@@ -0,0 +1,164 @@
+			if (oAuth.Exists(x => x.Name.Equals(oAuth2Name)))
+			{
+				if (oAuth.Exists(x => x.Name.Equals(oAuth2Name) && x.ContextUser == user.ArtifactID))
+				{
+					Console.WriteLine($"User ({user.EmailAddress}) already has OAuth2 Credentials within ({oAuth2Name})");
+					return oAuth.Find(x => x.Name.Equals(oAuth2Name) && x.ContextUser == user.ArtifactID);
+				}
+
+				throw new Exception($"A different user was has OAuth2 Credentials within ({oAuth2Name})");
+			}
+
+			Services.Security.Models.OAuth2Client oAuthResult = await _oAuth2ClientManager.CreateAsync(oAuth2Name, OAuth2Flow.ClientCredentials, new List<Uri>(), user.ArtifactID);
+			if (oAuthResult != null)
+			{
+				Console.WriteLine($"Created OAuth2 Client Credentials for ({user.EmailAddress}) within ({oAuthResult.Name})");
+				return oAuthResult;
+			}
+			else
+			{
+				throw new Exception($"Failed to create OAuth2 Client Credentials for ({user.EmailAddress}) within ({oAuth2Name})");
+			}
+		}
+
+		/// <summary>
+		/// Use this to get the Bearer Token that will give you access Relativity and various endpoints
+		/// </summary>
+		/// <param name="protocol"></param>
+		/// <param name="serverAddress"></param>
+		/// <param name="clientId"></param>
+		/// <param name="clientSecret"></param>
+		/// <param name="scope"></param>
+		/// <param name="grantType"></param>
+		/// <returns></returns>
+		public async Task<string> GetBearerTokenAsync(string protocol, string serverAddress, string clientId, string clientSecret, string scope = "SystemUserInfo", string grantType = "client_credentials")
+		{
+			string tokenUrl = GetTokenUrl(protocol, serverAddress);
+			string tokenResponse = await GetTokenAsync(tokenUrl, clientId, clientSecret, scope, grantType);
+
+			dynamic data = JObject.Parse(tokenResponse);
+			string bearerToken = data.access_token.Value;
+			return bearerToken;
+		}
+
+		/// <summary>
+		/// Use this to cleanup OAuth2 during tests
+		/// </summary>
+		/// <param name="clientId"></param>
+		/// <returns></returns>
+		public async Task DeleteOAuth2ClientAsync(string clientId)
+		{
+			await _oAuth2ClientManager.DeleteAsync(clientId);
+		}
+
+		/// <summary>
+		/// Retrieves User info by email
+		/// </summary>
+		/// <param name="email"></param>
+		/// <returns></returns>
+		private DTOs.User GetUser(string email)
+		{
+			_rsapiClient.APIOptions = new APIOptions { WorkspaceID = -1 };
+
+			DTOs.User user;
+
+			Condition userQueryCondition = new TextCondition(DTOs.UserFieldNames.EmailAddress, TextConditionEnum.EqualTo, email);
+
+			DTOs.Query<DTOs.User> userQuery = new DTOs.Query<DTOs.User>(DTOs.FieldValue.AllFields, userQueryCondition, new List<Sort>());
+
+			DTOs.QueryResultSet<DTOs.User> resultSet = _rsapiClient.Repositories.User.Query(userQuery);
+
+			if (resultSet.Success && resultSet.TotalCount > 0)
+			{
+				user = resultSet.Results.First().Artifact;
+			}
+			else
+			{
+				throw new Exception($"Could not find user {email}");

```

Comment:
can we retrieve the error message from the resultSet and throw it with the exception message. Will help in troubleshooting.